backend/logic/curriculums/curriculums.js

```diff
@@ -0,0 +1,95 @@
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format");
+
+var createCurriculum = async (name, type, department, numOfElectives, cores, techComps, comps) => {
+    let error = false;
+  
+    if (!format.verifyCurriculumName(name)) {
+      error = "Invalid curriculum name";
+    } else if(!format.verifyCurrType(type)){
+      error = "Invalid curriculum type";
+    } else if(!format.verifyDepartment(department)){
+      error = "Invalid department";
+    } else if(!format.verifyNumOfElectives(numOfElectives)) {
+      error = "Invalid number of electives"
+    }
+  
+    if (!error == false) {
+      console.error(error);
+      throw new Error(error);
+    }
+  
+    format.verifyCourse(cores);
+    format.verifyCourse(techComps);
+    format.verifyCourse(comps);
+  
+    let connection = await mysql.getNewConnection();
+  
+    try {
+      await connection.beginTransaction();
+      await connection.query("INSERT INTO curriculums (curriculum_name, type, department, numOfElectives) VALUES(?, ?, ?, ?);", [
```

Comment:
Just to confirm is this to create a curriculum for the user? If it is just to create a curriculum for a year for a program, there is a script that was made (just needs to be fine tuned and merged)

Reply:
It was to create a particular curriculum for a year which was to be done by admins. But we could also use it for creating a curriculum for a user for sure